graphviz/files.py

```diff
@@ -170,13 +171,14 @@ def render(self, filename=None, directory=None, view=False, cleanup=False):
             subprocess.CalledProcessError: If the exit status is non-zero.
             RuntimeError: If viewer opening is requested but not supported.
         """
-        filepath = self.save(filename, directory)
+        if cleanup:
+            filepath = self.save(tempfile.mkstemp('.gv')[1])
```

Comment:
I think one would prefer to use the file object from `tempfile.NamedTemporaryFile(delete=False)` within `save()`, see also https://www.logilab.org/blogentry/17873